modules/formulize/include/synchronization.php

```diff
@@ -0,0 +1,164 @@
+            $fileHandle = fopen($filePath, 'w');
+            fputcsv($fileHandle, $dataArray);
+            fclose($fileHandle);
+        }
+        
+        /*
+         * getTemplateFilePaths returns array containing all paths for template and custom_code files in Formulize directory
+         *
+         * return paths     string array containing paths for all template files
+         */
+        function getTemplateFilePaths(){
+            $screensPath = XOOPS_ROOT_PATH . "/modules/formulize/templates/screens";
+            $customCodePath = XOOPS_ROOT_PATH . "/modules/formulize/custom_code";
+            $paths = Array();
+            
+            // iterate $screenPath directory and store all file paths in $paths array
+            foreach (new RecursiveIteratorIterator(new RecursiveDirectoryIterator($screensPath)) as $filename){
+                if ($filename->isDir()) continue; // skip "." and ".."
+                array_push($paths, $filename);
+            }
+            
+            // iterate $customCodePath directory and store all file paths in $paths array
+            foreach (new RecursiveIteratorIterator(new RecursiveDirectoryIterator($customCodePath)) as $filename){
+                if ($filename->isDir()) continue; // skip "." and ".."
+                array_push($paths, $filename);
+            }
+            
+            return $paths;
+        }
+        
+        /*
+         * createArchive function used to create an archive (.zip) file and insert given files into it
+         *
+         * param archiveName        string representing path to new or existing zip file. path should have ".zip" extension
+         * param listOfFiles        string array containing 1 path for each file to be inserted into archive
+         * return archivePath       path to archive file
+         */
+        function createArchive($archiveName, $listOfFiles){
+            //$archivePath = "C:/xampp/htdocs/".$archiveName;
+            $archivePath = XOOPS_ROOT_PATH . "/modules/formulize/export"; // path where archive is created
+            
+            $zip = new ZipArchive(); // create ZipArchive object
+            
+            // open archive object. ".zip" file is only created once a file has been added to it
+            if ($zip->open($archivePath, ZIPARCHIVE::CREATE) !== TRUE) {
+                die ("Could not open archive");
+            }
+            
+            foreach($listOfFiles as $file){
+                $zip->addFile($file) or die ("ERROR: Could not add file: $file");
+            }
+            
+            $zip->close(); // close and save archive
+            
+            return $archivePath;
+        }
+
+
+        include_once "../class/PDO_Conn.php";
+
+        /*
+         * syncTablesList function returns a complete list of database tables that are required to be synced
+         */
+        function syncTablesList() {
```

Comment:
We need to add a few other tables:

groups
group_permission
xoopsnotifications